### ISSUE TITLE
Uncaught exception in library when handling exception with no stack trace

### ISSUE THREAD

**Issue by js-kyle** [microsoft/ApplicationInsights-node.js#489]
When promises are rejected without an error, the application crashes with the below Uncaught exception:

```
node_modules/applicationinsights/out/AutoCollection/Exceptions.js:34
                        throw THIS_IS_APPLICATION_INSIGHTS_RETHROWING_YOUR_EXCEPTION; // Error originated somewhere else in your app
                        ^

TypeError: Cannot read property 'stack' of undefined
    at Function.EnvelopeFactory.createExceptionData (/Users/kylemartin/Desktop/node_modules/applicationinsights/out/Library/EnvelopeFactory.js:132:40)
    at Function.EnvelopeFactory.createEnvelope (/Users/kylemartin/Desktop/node_modules/applicationinsights/out/Library/EnvelopeFactory.js:32:40)
    at NodeClient.TelemetryClient.track (/Users/kylemartin/Desktop/node_modules/applicationinsights/out/Library/TelemetryClient.js:102:44)
    at NodeClient.TelemetryClient.trackException (/Users/kylemartin/Desktop/node_modules/applicationinsights/out/Library/TelemetryClient.js:55:14)
    at AutoCollectExceptions.handle (/Users/kylemartin/Desktop/node_modules/applicationinsights/out/AutoCollection/Exceptions.js:30:35)
    at process.emit (events.js:182:13)
    at emitPromiseRejectionWarnings (internal/process/promises.js:119:20)
    at process._tickCallback (internal/process/next_tick.js:69:34)
    at Function.Module.runMain (internal/modules/cjs/loader.js:744:11)
    at startup (internal/bootstrap/node.js:285:19)
```

This can be reproduced on `1.2.0` as well as older versions, by:

test.js
```
let appInsights = require('applicationinsights');
appInsights.setup().start();

Promise.reject();
```

`APPINSIGHTS_INSTRUMENTATIONKEY="string" node test.js`

**Comment by OsvaldoRosado** [comment]
Thanks for the report! It looks like we're also unintentionally suppressing important Node warnings from this scenario. For example, when I run this sample without the AI SDK, Node shows this warning (though does not crash).

```
(node:1808) UnhandledPromiseRejectionWarning: undefined
(node:1808) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 1)
(node:1808) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
```

**Comment by ta-run** [comment]
Hey @OsvaldoRosado is this bug addressed with `1.5.0`? The referenced issue is merged but the bug is still open.

I'm currently on `v1.4.1` and get the `Cannot read property 'stack' of undefined` error.
If this is not resolved, is there a workaround for this? Thanks :)

**Comment by OsvaldoRosado** [comment]
@markwolff was this fixed in newer releases?

I understand Node itself warns that it will crash for unhandled promises in newer versions, but we probably shouldn't crash on Node's behalf :smile:

**Comment by markwolff** [comment]
I don't think this has been addressed yet. I'll include a fix for it in next week's release (`1.6.0`)
